app/socket-apis/upsert-rank.js

```diff
@@ -0,0 +1,55 @@
+//https://github.com/EnCiv/civil-pursuit/issues/211
+import { ObjectId } from 'mongodb'
+
+const Joi = require('joi')
+const Ranks = require('../models/ranks')
+
+// Define the validation schema using Joi
+const rankSchema = Joi.object({
+  _id: Joi.string().optional(), // _id is optional when creating a new document
+  stage: Joi.string().valid('pre', 'post', 'why').required(), // stage must be one of 'pre', 'post', 'why'
+  category: Joi.string().valid('most', 'least', 'neutral').required(), // category must be one of 'most', 'least', 'neutral'
+  parentId: Joi.string().required(), // parentId is required
+  userId: Joi.string().optional(), // userId is optional and will be deleted later if needed
+  discussionId: Joi.string().required(), // discussionId is required
+  round: Joi.number().integer().min(1).required(), // round must be an integer >= 1
```

Comment:
please allow round to be 0.  There may be  an issue / difference between the internal representation and what is shown to the users..

Reply:
Already change the round to be >= 0